attendance/views.py

```diff
@@ -43,6 +39,27 @@ def attendance(request):
     return render(request, "attendance.html", context)
 
 
+@group_required_multiple(["staff", "teacher"])
+def take_attendance(request):
+    context = take_attendance_context(
+        request.GET.get("course_id"),
+        get_date_from_template_returned_string(request.GET.get("date")),
+    )
+    return render(request, "attendance.html", context)
+
+
+@group_required_multiple(["staff", "teacher"])
+def notify_absent_students(request):
+    # This method is a candidate for an async_task
+    date = get_date_from_template_returned_string(request.GET.get("date"))
+    course = Classroom.objects.get(id=request.GET.get("course_id"))
+    absences = Attendance.objects.filter(date=date, classroom_id=course.id, presence="Absent")
+    student_list = list(DjangoUser.objects.filter(id__in=[absence.student_id for absence in absences]))
```

Comment:
`student_list` is not guaranteed to be in the same order as the query used for `absences`, you probably want to use [select_related()](https://docs.djangoproject.com/en/2.2/ref/models/querysets/#select-related) on the above absences query instead of doing this query at all.

Reply:
Resolved in 989e97ca4de16df31e508a8eeb5154ade44b3bd5